pkg/controller/installation/core_controller.go

```diff
@@ -1177,6 +1180,22 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 			nodeTerminating = true
 		}
 	}
+
+	// Determine the felix health port to use. Prefer the configuration from FelixConfiguration,
+	// but default to 9099.
+	felixHealthPort := 9099
+	if felixConfiguration.Spec.HealthPort != nil {
```

Comment:
I think you're still probably working/thinking about things for this PR but I would consider defaulting the felixConfiguration HealthPort and writing that back to the datastore.
Do you see any issue with doing that?